### DISCUSSION TITLE
how can you create languages with level in laravel backpack?

### DISCUSSION THREAD

**Discussion by Bekshakx** [Laravel-Backpack/community-forum#264] [Q&A (Help)] [answered]
My "languages" column
[image][1]

I want that when I click save, there should be several records in the database.

I want to create one action, many entries in the database.
For example:

I write the name of the language

[![enter image description here][2]][2]

database:

id | title | level

1 | English | A1 - Primary

2 | English | A2 - Elementary

3 | English | B1 - Medium

  [1]: https://i.stack.imgur.com/mTT5T.png
  [2]: https://i.stack.imgur.com/EPIwk.png

**Comment by Bekshakx** [comment]
@tabacitu  help me please

**Comment by pxpm** [comment]
Hello @Bekzhan99 
I'd recommend you use the model events, so when the model is saved/saving you can create the aditional records you need. 
Either use the the Laravel way, or if you are a fan of having all crud definition inside the controller you can use them directly in the controller: https://backpackforlaravel.com/docs/5.x/release-notes#use-model-events-inside-crudcontrollers

Cheers

**Comment by Bekshakx** [comment]
@pxpm  can you give examples? I'm new and I don't understand

**Comment by pxpm** [comment]
Well @Bekzhan99  I recommend you start by learning the basics: https://laravel.com/docs/9.x/eloquent#events , so basically what you want is listening to the `saving` event and add the extra fields to the model before it's saved in database. 

If your process is more complex you can totally overwrite the crud save process by overwritting the crud store/update functions: https://backpackforlaravel.com/docs/5.x/getting-started-crud-operations#callbacks-1 that way you have total control on what is going to be created/updated, create related models yourself etc etc. 

Sorry can't be much more help here, this should be straigthfoward to follow.

**Reply by Bekshakx** [reply]
@pxpm  Thank you!

**Reply by Bekshakx** [reply]
@pxpm 
CRUDController:
![image](https://user-images.githubusercontent.com/47516570/161690375-a799abc8-2e17-4ea0-8ade-cb648db17e25.png)
Language Model:
![image](https://user-images.githubusercontent.com/47516570/161690400-21893f8c-23f1-4fc4-8c96-985067dba80b.png)
LanguageListener:
![image](https://user-images.githubusercontent.com/47516570/161690556-05707109-26e5-4666-8ea5-753f4fe16b79.png)


I call the event but there are many cycles why?

**Comment by pxpm** [comment, marked as answer]
@Bekzhan99 you are using two distinct events types, you can use any of them, but both of them is overkill I think. 

From your example, if you want is to the user to be able to select a language and a level for that language, I think you should be doing the following: 

Create a table with all the languages (create a migration and a seeder to seed all the languages in a table in the database). Create a new table where you will hold the relation with user/languages like `languages_user` with columns: `user_id, language_id, level`. Configure the relations in user and language models, they should be `belongsToMany`. Use a `repeatable` field in the user controller (or where you want to select the languages/levels), the name of the repeatable field is the relation name, and add "level" as a subfield (could be a select from array or something). https://backpackforlaravel.com/docs/5.x/crud-how-to#belongstomany-n-n-relationship

In the end the user should be able to select one or more languages and for each one choose a level. No need for events or anything extra.

Cheers
